datastore/__init__.py

```diff
@@ -1,8 +1,46 @@
-__version__ = '0.3.6'
-__author__ = 'Juan Batiz-Benet'
-__email__ = 'juan@benet.ai'
+"""
+Datastore is a generic layer of abstraction for data store and database access.
+It is a **simple** API with the aim to enable application development in a
+datastore-agnostic way, allowing datastores to be swapped seamlessly without
+changing application code. Thus, one can leverage different datastores with
+different strengths without committing the application to one datastore
+throughout its lifetime.
+"""
 
-from .core import *
-from .filesystem import *
+__version__ = "0.3.6"
+__author__ = "Juan Batiz-Benet, Alexander Schlarb"
+__email__ = "juan@benet.ai, alexander@ninetailed.ninja"
+__all__ = (
+	"Key", "Namespace",
+	"BinaryNullDatastore", "BinaryDictDatastore",
+	"ObjectNullDatastore", "ObjectDictDatastore",
+	"Query", "Cursor",
+	"SerializerAdapter",
+	
+	"abc", "typing", "util"
+)
 
+
+# import core.key
+from .core.key import Key
+from .core.key import Namespace
+
+# import core.binarystore, core.objectstore
+from .core.binarystore import NullDatastore as BinaryNullDatastore
+from .core.binarystore import DictDatastore as BinaryDictDatastore
+
+from .core.objectstore import NullDatastore as ObjectNullDatastore
+from .core.objectstore import DictDatastore as ObjectDictDatastore
+
```

Comment:
We will also have to expose FileSystemDatastore and FileReader

Reply:
My thought was that the filesystem datastore has to be explicitly imported using `datastore.filesystem` as it is not needed if another datastore (like leveldb or the in-memory dict one) is used.

`FileReader` is just an implementation detail of how the datastore manages to return its `datastore.abc.ReceiveStream`. Its not really meant to be used directly.

Reply:
Okay 👍